app/api.go

```diff
@@ -293,6 +293,40 @@ func (api *API) latestHandler(r *http.Request) (interface{}, error) {
 	}, nil
 }
 
+type getLeafResponse struct {
+	Leaf *trillian.GetLeavesByIndexResponse
+	Key  []byte
+}
+
+func (api *API) getleafHandler(r *http.Request) (interface{}, error) {
+	leafSizeInt := int64(0)
+	leafIndex := r.URL.Query().Get("leafindex")
+
+	// error check leaf index
+
+	if leafIndex != "" {
+		var err error
+		leafSizeInt, err = strconv.ParseInt(leafIndex, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	resp, err := api.logClient.GetLeavesByIndex(r.Context(), &trillian.GetLeavesByIndexRequest{
+		LogId:     api.tLogID,
+		LeafIndex: []int64{leafSizeInt},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	logging.Logger.Info(resp)
+
+	return getLeafResponse{
+		Leaf: resp,
```

Comment:
I think you need to also include the public key here. It's on `api.pubkey.Der`. Then in the client you can verify with that.